static/app/actionCreators/pageFilters.tsx

```diff
@@ -231,46 +246,45 @@ export function updateProjects(
 
   PageFiltersActions.updateProjects(projects, options?.environments);
   updateParams({project: projects, environment: options?.environments}, router, options);
-}
-
-function isProjectsValid(projects: ProjectId[]) {
-  return Array.isArray(projects) && projects.every(project => isInteger(project));
+  persistPageFilters(options);
 }
 
 /**
- * Updates store and global datetime selection URL param if `router` is supplied
+ * Updates store and updates global environment selection URL param if `router` is supplied
```

Comment:
This diff is a bit tricky to read, since I moved the `updateEnvironments` function up above the updateDateTime :/